feat(Toolbar): new component

mrkaashee opened the pull request:
<!---
☝️ PR title should follow conventional commits (https://conventionalcommits.org)
-->

### 🔗 Linked issue

<!-- If it resolves an open issue, please link the issue here. For example "Resolves #123" -->

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->
I've added a new `<Toolbar>` component. 

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

mrkaashee committed:
feat(Toolbar): new component

mrkaashee committed:
test: add Toolbar snapshot tests

ineshbose commented:
Is this similar to DashboardToolbar?

mrkaashee commented:
> Is this similar to DashboardToolbar?

It’s similar in purpose but more general. Toolbar is reusable beyond dashboards and includes more functionality like variant support and a title prop. I also plan to extend it further in future updates.

ineshbose commented:
Yes I figure the layout is the same, and to be honest I noticed that the DashboardToolbar isn't specific to Dashboard components within a DashboardGroup as it doesn't inject any of the dashboard context (I am using it in my own app outside of Dashboard), therefore I also feel its generic and good to introduce to non-pro * imo *

mrkaashee commented:
Agreed — makes sense. This version aims to be more flexible and generic .

benjamincanac reviewed (commented):
Why not take advantage of Reka UI's Toolbar component? 🤔
https://reka-ui.com/docs/components/toolbar

mrkaashee commented:
Thanks **@benjamincanac!** I considered Reka’s Toolbar, but since Nuxt UI already uses a custom one in `DashboardToolbar`, I followed the same approach.

Reka UI’s Toolbar is quite structured, but I wanted something more flexible.

This version is something I’ve already been using in my projects, so I decided to bring it into Nuxt UI with improvements like slots, variants, and better reusability.

mrkaashee committed:
Add size variants to Toolbar component

Introduces a new 'size' prop and size variants (sm, md, lg, xl) to the Toolbar component, updates theme definitions, playground usage, and tests to support and demonstrate the new sizing options.

benjamincanac commented:
The `DashboardToolbar` from Nuxt UI Pro is a layout component that is meant to be used under a `DashboardNavbar` to build a dashboard, basically it's just a div with `flex items-center justify-between` and some slots. It is different from a functional Toolbar component like https://reka-ui.com/docs/components/toolbar

I guess we can close this if you don't plan to implement it on top of Reka UI? Also, I'd like to implement the Toggle/ToggleGroup components before this one: https://reka-ui.com/docs/components/toggle-group

kashiarain commented:
Thanks for the clarification!

No need to close this PR — I will implement the Toolbar on top of Reka UI.
Will update this PR once ready!
